From 79e8e62f662cab166fb8d28288f40d5863b52114 Mon Sep 17 00:00:00 2001
From: Janne Grunau <j@jannau.net>
Date: Fri, 9 Oct 2026 18:07:44 +0200
Subject: [PATCH] display: Limit M1/M2 Sequoia DCP power-cycle to M1/M2

The `!has_notch` check doesn't exclude 24-inch iMacs with M3 and later.

Signed-off-by: Janne Grunau <j@jannau.net>
---
 src/display.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/display.c b/src/display.c
index de53a398c..b7d26dec3 100644
--- a/src/display.c
+++ b/src/display.c
@@ -658,7 +658,7 @@ int display_init(void)
         fb_clear_direct(); // Old m1n1 stage1 ends up with an ugly logo situation, clear it.
         return display_configure(NULL);
 #ifndef CHAINLOADING
-    } else if (!has_notch && firmware_sfw_in_range(V15_0B1, FW_MAX) &&
+    } else if (!has_notch && chip_id <= T8112 && firmware_sfw_in_range(V15_0B1, FW_MAX) &&
                os_firmware.version < V15_0B1) {
         printf("display: Internal display on t8103 or t8112 with Sequoia SFW, power cycling\n");
         display_needs_power_cycle = true;
